engine/orchestration/src/com/cloud/agent/manager/AgentManagerImpl.java

```diff
@@ -821,9 +819,9 @@ protected boolean handleDisconnectWithInvestigation(final AgentAttache attache,
             try {
                 nextStatus = host.getStatus().getNextStatus(event);
             } catch (final NoTransitionException ne) {
-                /* Agent may be currently in status of Down, Alert, Removed, namely there is no next status for some events.
-                 * Why this can happen? Ask God not me. I hate there was no piece of comment for code handling race condition.
-                 * God knew what race condition the code dealt with!
+                /*
```

Comment:
@miguelaferreira  If you extract the code from lines 819-827 to a method, you could add that comment block to it as a Java doc, a method such as “getHostNextStatus” would do. I would only suggest you to remove the sarcastic bit of it.